packages/flutter/lib/src/painting/borders.dart

```diff
@@ -30,16 +30,91 @@ enum BorderStyle {
 /// - [inside] provides padding with full [BorderSide.width].
 /// - [center] provides padding with half [BorderSide.width].
 /// - [outside] provides zero padding, as stroke is drawn entirely outside.
-enum StrokeAlign {
+@immutable
+class StrokeAlign {
```

Comment:
I don't know, I only reused what existed already. Ideally, it would be StrokeAlign.inside/center/outside/custom, but enum doesn't support custom.

Like, how would you initialize it in your current proposal?  Something like this?

<img width="687" alt="image" src="https://user-images.githubusercontent.com/351125/181121095-33097888-f824-4237-95b2-c5b9c17c750b.png">

Reply:
Yep, exactly (although I'd still advocate -1 to 1 instead).

Reply:
Just answer the other comment, how to lerp the borders, my brain is too melted for that lol

If you can get that, I'll get the -1, 0, 1.

Reply:
I hate the new names. I think we can stay with the double, but I propose something in the middle (which can be confusing, open to discussion): 
<img width="816" alt="image" src="https://user-images.githubusercontent.com/351125/181125004-8c710fa8-f45f-4c15-ba44-0a48b6ba4d9e.png">

`StrokeAlign.inside` which is just a shortcut.
Because `BorderSide.strokeAlignInside` seems too hidden for me.

Reply:
Update: what if StrokeAlign is closed and you can only call StrokeAlign.inside/center/outside which would work exactly like an enum except it gives you a double?